tests/testthat/test-frgtracker.R

```diff
@@ -10,11 +10,81 @@ test_that("multiplication works", {
 #function2 end
 
 #function3 start
+test_that("The input of generate_course_statistics is not valid", {
+  course_ids <- c("530", "540")
+  expect_error(generate_course_statistics(course_ids))
+
+  course_ids <- c(511)
+  expect_error(generate_course_statistics(course_ids))
+})
+
+test_that("The output of generate_course_statistics is incorrect", {
+  course_ids <- c("511", "522")
+  grade_511 <- c(84.66, 88.34, 87.66, 90.82)
+  grade_522 <- c(95.52, 87.92, 88.92, 92.80)
+
+  output <- as.numeric(generate_course_statistics(course_ids)[1,2])
```

Comment:
It is better to call the function once, store it in an object, and extracting data from the same object multiple times